test/test_variant_utils.py

```diff
@@ -0,0 +1,134 @@
+        with self.mock_key_manager():
+            vu = VariantUtils(env_name='cgap-dummy')
+            mock_gene = 'GENE'
+            mock_get_rare_variants_by_gene.return_value = (returned_variants * expected_length)
+            result = vu.create_dict_of_mutations(mock_gene)
+            if expected_length >= 10:
+                expected_result = {mock_gene: {100000: expected_length}}
+            else:
+                expected_result = {mock_gene: {}}
+            assert result == expected_result
+            mock_get_rare_variants_by_gene.assert_called_once_with(gene=mock_gene, sort='variant.ID')
+
+    @patch('dcicutils.variant_utils.VariantUtils.return_json')
+    def test_create_list_of_msa_genes(self, mock_return_json):
+        with self.mock_key_manager():
+            vu = VariantUtils(env_name='cgap-dummy')
+            mock_return_json.return_value = [
+                {'gene_symbol': 'GENE1', 'gene_summary': '...nerv...'},
+                {'gene_symbol': 'GENE2', 'gene_summary': '..........'},
+                {'gene_symbol': 'GENE3', 'gene_summary': '...neur...'}
+            ]
+            result = vu.create_list_of_msa_genes()
+            expected_result = ['GENE1', 'GENE3']
+            assert result == expected_result
+            mock_return_json.assert_called_once_with('gene.json')
+
+    @patch('dcicutils.variant_utils.VariantUtils.get_rare_variants_by_gene')
+    def test_find_number_of_sample_ids(self, mock_get_rare_variants_by_gene):
+        with self.mock_key_manager():
+            vu = VariantUtils(env_name='cgap_dummy')
+            mock_gene = 'GENE'
+            mock_get_rare_variants_by_gene.return_value = [
+                {'CALL_INFO': 'ABC123'},
+                {'CALL_INFO': 'ABC123'},
+                {'CALL_INFO': 'BCD234'},
+                {'CALL_INFO': 'CDE345'}
+            ]
+            result = vu.find_number_of_sample_ids(mock_gene)
+            expected_result = 3
+            assert result == expected_result
+            mock_get_rare_variants_by_gene.assert_called_once_with(gene=mock_gene, sort='variant.ID')
+
+    @pytest.mark.parametrize('pos', [
+        '100000',
+        '200000',
+        '300000',
+        '400000'
+    ])
+    @patch('dcicutils.variant_utils.VariantUtils.create_dict_from_json_file')
+    def test_create_url(self, mock_create_dict_from_json_file, pos):
+        with self.mock_key_manager():
+            vu = VariantUtils(env_name='cgap_dummy')
+            mock_gene = 'GENE'
+            mock_create_dict_from_json_file.return_value = {
+                'GENE': {pos: 20, '123456': 10},
+                'OTHER_GENE': {pos: 10}
+            }
+            result = vu.create_url(gene=mock_gene)
+            expected_result = vu.SEARCH_RARE_VARIANTS_BY_GENE + mock_gene + ('&variant.POS.from={pos}\
+                                                                             &variant.POS.to={pos}&sort=-DP')
```

Comment:
Likely to be much cleaner with f-strings ie:
```
expected_result = (f'{vu.SEARCH_RARE_VARIANTS_BY_GENE}{mock_gene}'
                   f'&variant.POS.from={pos}'
                   f'&variant.POS.to={pos}&sort=-DP')
```

It's also not clear this is working as desired, seems like you may want pos actually injected and that isn't happening as written I don't think?